discv5/discv5-wire-test-vectors.md

```diff
@@ -1,339 +1,156 @@
+    4539717307a0208cd208d65093ccab5aa596a34d7511401987662d8cf62b1394
+    71
 
 ## Cryptographic Primitives
 
-This section provides test vectors for the currently supported "v4" identity
-scheme.
+This section provides test vectors for the currently supported "v4" identity scheme.
 
 ### ECDH
 
-The ECDH function takes the elliptic-curve scalar multiplication of a public
-key and a private key. The wire protocol describes this process.
-
-The input public key is an uncompressed secp256k1 key (64 bytes) and the
-private key is a raw secp256k1 private key (32 bytes).
-
-#### Input Parameters
-
-    public-key: 0x9961e4c2356d61bedb83052c115d311acb3a96f5777296dcf297351130266231503061ac4aaee666073d7e5bc2c80c3f5c5b500c1cb5fd0a76abbb6b675ad157
-    secret_key: 0xfb757dc581730490a1d7a00deea65e9b1936924caaea8f44d476014856b68736
+The ECDH function takes the elliptic-curve scalar multiplication of a public key and a
+private key. The wire protocol describes this process.
 
-#### Expected Output
+    public-key = 0x039961e4c2356d61bedb83052c115d311acb3a96f5777296dcf297351130266231
+    secret-key = 0xfb757dc581730490a1d7a00deea65e9b1936924caaea8f44d476014856b68736
 
-This output is the result of the ECDH function which will be used as input to
-the HKDF-EXTRACT and HKDF-EXPAND functions.
+This output is the result of the ECDH function which will be used by the KDF.
 
-    shared-secret: 0x033b11a2a1f214567e1537ce5e509ffd9b21373247f2a3ff6841f4976f53165e7e
+    shared-secret = 0x033b11a2a1f214567e1537ce5e509ffd9b21373247f2a3ff6841f4976f53165e7e
 
 ### Key Derivation
 
-This test vector takes a secret key (as calculated from the previous test
-vector) along with two node id's and an `id-nonce`. This demonstrates the
-HKDF-EXPAND and HKDF-EXTRACT functions using the added key-agreement string as
-described in the wire specification.
-
-Given a secret key (calculated from ECDH above) two `node-id`s (required to
-build the `info` as described in the specification) and the `id-nonce`
-(required for the HKDF-EXTRACT function), this should produce an
-`initiator-key`, `recipient-key` and an `auth-resp-key`.
-
-#### Input Parameters
-
-    secret-key: 0x02a77e3aa0c144ae7c0a3af73692b7d6e5b7a2fdc0eda16e8d5e6cb0d08e88dd04
-    node-id-A: 0xa448f24c6d18e575453db13171562b71999873db5b286df957af199ec94617f7
-    node-id-B: 0x885bba8dfeddd49855459df852ad5b63d13a3fae593f3f9fa7e317fd43651409
-    id-nonce: 0x0101010101010101010101010101010101010101010101010101010101010101
+This test vector checks the complete key derivation as used by the handshake.
 
-#### Expected Outputs
+    ephemeral-key = 0xfb757dc581730490a1d7a00deea65e9b1936924caaea8f44d476014856b68736
+    dest-pubkey = 0x0317931e6e0840220642f230037d285d122bc59063221ef3226b1f403ddc69ca91
+    node-id-a = 0xaaaa8419e9f49d0083561b48287df592939a8d19947d8c0ef88f2a4856a69fbb
+    node-id-b = 0xbbbb9d047f0488c0b5a93c1c3f2d8bafc7c8ff337024a55434a0d0555de64db9
+    whoareyou.masking-iv = 0x01010101010101010101010101010101
```

Comment:
Forgot to update these?
I managed to match your results with instead:
 `challenge-data = 0x000000000000000000000000000000006469736376350001010102030405060708090a0b0c00180102030405060708090a0b0c0d0e0f100000000000000000`

Reply:
Thanks! Yes, I pasted in the new results and wanted to delete the old text but deleted the new text instead.